Migration.Toolkit.CLI/ConfigurationValidator.cs

```diff
@@ -0,0 +1,110 @@
+using System.Diagnostics.Contracts;
+using Microsoft.Extensions.Configuration;
+using Migration.Toolkit.Common;
+using Migration.Toolkit.Common.Helpers;
+using Migration.Toolkit.Core.Helpers;
+
+namespace Migration.Toolkit.CLI;
+
+public enum ValidationMessageType
+{
+    Error,
+    Warning,
+}
+
+public record ValidationMessage(ValidationMessageType Type, string Message, string? RecommendedFix = null);
+
+public static class ConfigurationValidator
+{
+    [Pure]
+    public static IEnumerable<ValidationMessage> GetValidationErrors(IConfigurationRoot root)
+    {
+        var settings = root.GetSection("Settings");
+
+        if (settings is null)
+        {
+            yield return new ValidationMessage(ValidationMessageType.Error, "Section 'Settings' is required");
+        }
+        
+        if (CheckCfgValue(settings?.GetValue<string>("SourceConnectionString")))
```

Comment:
Please, use nameof(ToolkitConfiguration.SourceConnectionString) instead of hardcoded text. The same approach for all checks.